Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_PlayerPresence.cs

```diff
@@ -17,18 +28,66 @@ public class VRTK_PlayerPresence : MonoBehaviour
         private float crouchMargin = 0.5f;
         private float lastPlayAreaY = 0f;
 
+        private bool isFalling = false;
+
+        public virtual void OnPresenceFallStarted(PlayerPresenceEventArgs e)
+        {
+            if (PresenceFallStarted != null)
+            {
+                PresenceFallStarted(this, e);
+            }
+        }
+
+        public virtual void OnPresenceFallEnded(PlayerPresenceEventArgs e)
+        {
+            if (PresenceFallEnded != null)
+            {
+                PresenceFallEnded(this, e);
+            }
+        }
+
+        private PlayerPresenceEventArgs SetPlayerPhysicsEvent(float fallDistance)
```

Comment:
Private class should be at bottom of stack, group classes public together, then protected, then private